src/Catel.Core/Catel.Core.Shared/IoC/ServiceLocator.cs

```diff
@@ -40,39 +40,28 @@ public RegisteredInstanceInfo(ServiceLocatorRegistration registration, object in
         [DebuggerDisplay("{Type} ({Tag})")]
         private class ServiceInfo
         {
-            private int _hash;
+            private readonly int _hash;
 
             #region Constructors
             public ServiceInfo(Type type, object tag)
             {
                 Type = type;
                 Tag = tag;
+                _hash = HashHelper.CombineHash(Type.GetHashCode(), Tag != null ? Tag.GetHashCode() : 0);
```

Comment:
Main purpose of this class is to be dictionary key. I don't know if it is created without being key. I think that lazy computation of hash-code makes this code more complex and is always delayed never prevented.